### PR TITLE
[GEOS-10741] Remove deprecated YUI usage

### PR DESCRIPTION
**Pull request by taba90**
[GEOS-10741](https://osgeo-org.atlassian.net/browse/GEOS-10741)

This pr:
- remove wicket date time dependencies
- provides new components where needed namely: for datepicker in monitor and metadata modules; for Date Label in wicket core (System status).

Below a video showing the new datepicker.

https://user-images.githubusercontent.com/46672505/200799679-22e0eb73-87bf-435e-83b2-c1192d4979ad.mp4

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/web/core/src/main/java/org/geoserver/web/wicket/DatePicker.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+// below some variable and functions to map date format tokens from java to moment js eg. d -> D for day of month.
```
Comment (aaime) [inline src/web/core/src/main/java/org/geoserver/web/wicket/DatePicker.js:1]: This is something you added ? Should have the GeoServer license header then.

**Review by aaime — CHANGES_REQUESTED** [review]
Overall looks good, just noting that we often put bits of javascript coming from other libraries in a dedicated "js" subfolder:

https://github.com/geoserver/geoserver/tree/main/src/web/core/src/main/java/org/geoserver/web/wicket/js

At the same time, bridge javascript bits that we write ourselves are on the side of the Java class using them, instead.

File: src/web/core/src/main/java/org/geoserver/web/wicket/DatePicker.js (hunk 1 above)
Comment (taba90) [reply src/web/core/src/main/java/org/geoserver/web/wicket/DatePicker.js:6]: fixed

**Comment by taba90** [comment]
@aaime thanks for the review. Feedback applied.

**Pull request merged**